main.go

```diff
@@ -0,0 +1,325 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/pingcap/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type scrapeConfig []struct {
+	HonorTimestamps bool   `yaml:"honor_timestamps"`
+	JobName         string `yaml:"job_name"`
+	MetricsPath     string `yaml:"metrics_path"`
+	Params          struct {
+		Module []string `yaml:"module"`
+	} `yaml:"params"`
+	RelabelConfigs []struct {
+		SourceLabels []string `yaml:"source_labels,omitempty"`
+		TargetLabel  string   `yaml:"target_label,omitempty"`
+		Replacement  string   `yaml:"replacement,omitempty"`
+	} `yaml:"relabel_configs"`
+	Scheme         string `yaml:"scheme"`
+	ScrapeInterval string `yaml:"scrape_interval"`
+	ScrapeTimeout  string `yaml:"scrape_timeout"`
+	StaticConfigs  []struct {
+		Targets []string `yaml:"targets"`
+		Labels  struct {
+			Module string `yaml:"module"`
+		} `yaml:"labels"`
+	} `yaml:"static_configs"`
+}
+
+type environmentVariables struct {
+	PublicHostedZoneID   string
+	PrivateHostedZoneID  string
+	PrometheusNamespace  string
+	PrometheusSecretName string
+	MattermostAlertsHook string
+	ExcludedTargets      []string
+	AdditionalTargets    []string
+	DevMode              string
+}
+
+func main() {
+	envVars, err := validateAndGetEnvVars()
+	if err != nil {
+		log.WithError(err).Error("Environment variable validation failed")
+		err = sendMattermostErrorNotification(err, "Environment variable validation failed")
+		if err != nil {
+			log.WithError(err).Error("Failed to send Mattermost error notification")
+		}
+		return
```

Comment:
even if there is an error we don't care?

maybe we use the os.exit(0) when success and os.Exist(1) or another number when fail?